frontend-opal/src/components/PortfolioContainer/PortfolioContainer.js

```diff
@@ -11,11 +11,21 @@ import AllocationInfo from './AllocationInfo';
 import Instruments from './Instruments';
 
 const PortfolioContainer = ({ fetchPortfolio }) => {
-  useEffect(() => fetchPortfolio(), [fetchPortfolio]);
+  useEffect(() => {
+    fetchPortfolio();
+  }, [fetchPortfolio]);
+
+  const [refreshing, setRefreshing] = React.useState(false);
+
+  const onRefresh = React.useCallback(async () => {
+    setRefreshing(true);
+    await fetchPortfolio();
+    setRefreshing(false);
+  }, [fetchPortfolio]);
```

Comment:
@WenyuD this is basically the same as the useEffect and the loading state of the data
we can leave it like this, but this is duplicated logic